index.bs

```diff
@@ -2222,6 +2259,84 @@ To <dfn>parse trigger specs</dfn> given a [=map=] |map|, a [=moment=]
+1. If |pendingSource|'s [=attribution source/attribution scope limit=] is null, return.
+1. Let |cache| be the user agent's [=attribution source cache=].
+1. [=set/iterate|For each=] |source| of |cache|:
+    1. Let |commonDestinations| be the [=set/intersection=] of |source|'s [=attribution source/attribution destinations=] and |pendingSource|'s [=attribution source/attribution destinations=].
+    1. If |commonDestinations| [=set/is empty=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/source origin=] and |pendingSource|'s [=attribution source/source origin=] are not [=same origin=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/attribution scope limit=] is null, [=iteration/continue=].
+    1. If |source|'s [=attribution source/attribution scope limit=] is greater than or equal to |pendingSource|'s [=attribution source/attribution scope limit=], [=iteration/continue=].
+    1. [=set/Remove=] |source| from |cache|.
+
+To <dfn>remove attribution scopes for sources with attribution scopes not selected if necessary</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |attributionScopeLimit| be |pendingSource|'s [=attribution source/attribution scope limit=].
+1. If |attributionScopeLimit| is null, return.
+1. Let |cache| be the user agent's [=attribution source cache=].
+1. [=list/iterate|For each=] |destination| in |pendingSource|'s [=attribution source/attribution destinations=]:
+    1. Let |matchingSources| be a new [=list/is empty|empty=] [=list=].
+    1. [=set/iterate|For each=] |source| of |cache|:
+        1. If |source|'s [=attribution source/source origin=] and |pendingSource|'s [=attribution source/source origin=] are not [=same origin=], [=iteration/continue=].
+        1. If |source|'s [=attribution source/attribution destinations=] does not [=set/contains|contain=] |destination|, [=iteration/continue=].
+        1. If |source|'s [=attribution source/attribution scopes=] is null or empty, [=iteration/continue=].
+        1. [=list/Append=] |source| to |matchingSources|.
+    1. Set |matchingSources| to the result of [=list/sort in descending order|sorting=] |matchingSources|
+        in descending order, with |a| being less than |b| if
+        |a|'s [=attribution source/source time=] is less than |b|'s [=attribution source/source time=].
+    1. Let |selectedAttributionScopes| be a [=set=].
+    1. If |pendingSource|'s [=attribution source/attribution scopes=] is not null:
+        1. Set |selectedAttributionScopes| to |pendingSource|'s [=attribution source/attribution scopes=].
+    1. [=set/iterate|For each=] |source| of |matchingSources|:
+        1. Let |newSelectedAttibutionScopes| be the [=set/union=] of |selectedAttributionScopes| and |source|'s [=attribution source/attribution scopes=].
+        1. If |newSelectedAttibutionScopes|'s [=set/size=] is greater than  |attributionScopeLimit|:
```

Comment:
Consider the following scenario for attribution scope limit = 3:
S1: attribution scopes: [0]
S2: attribution scopes: [1, 2]
S3: attribution scopes: [3, 4]

With the current algorithm, at S3, S2 would be deleted and S1 would be kept. Is this the intended behavior?

Reply:
Yes. that's what I had in mind. 

Alternatively we can remove scopes for sources before S2 as well. 

@giladbarkan-github @vikassahu29

Reply:
It seems to be a bit confusing as the scopes 1,2 were introduced later than 0.

Reply:
I think you are right, we should remove all sources before S2.

Reply:
resolved, please take a look at the latest snapshot.

Reply:
If S0 has scope [3], then yes, it should not be removed.

I would think we find the latest K attribution scopes, including those in the new source, and remove scopes that are not in the set.

Reply:
Per discussion offline, the key rule we want to apply is `all scopes or none are considered for any one source`.
in your scenario none of the scopes from S2 would be selected.